File: examples/swift-server/Sources/AppwriteModels/Tag.swift

/// Tag
public class Tag {

    /// Tag ID.
    public let id: String

    /// Function ID.
    public let functionId: String

    /// The tag creation date in Unix timestamp.
    public let dateCreated: Int

    /// The entrypoint command in use to execute the tag code.
    public let command: String

    /// The code size in bytes.
    public let size: String

    init(
        id: String,
        functionId: String,
        dateCreated: Int,
        command: String,
        size: String
    ) {
        self.id = id
        self.functionId = functionId
        self.dateCreated = dateCreated
        self.command = command
        self.size = size
    }

    public static func from(map: [String: Any]) -> Tag {
        return Tag(
            id: map["$id"] as! String,
            functionId: map["functionId"] as! String,
            dateCreated: map["dateCreated"] as! Int,
            command: map["command"] as! String,
            size: map["size"] as! String
        )
    }

    public func toMap() -> [String: Any] {
        return [
            "$id": id as Any,
            "functionId": functionId as Any,
            "dateCreated": dateCreated as Any,
            "command": command as Any,
            "size": size as Any
        ]
    }
                        
}

File: examples/flutter/lib/src/models/tag.dart
part of appwrite.models;

/// Tag
class Tag {
    /// Tag ID.
    final String $id;
    /// Function ID.
    final String functionId;
    /// The tag creation date in Unix timestamp.
    final int dateCreated;
    /// The entrypoint command in use to execute the tag code.
    final String command;
    /// The code size in bytes.
    final String size;

    Tag({
        required this.$id,
        required this.functionId,
        required this.dateCreated,
        required this.command,
        required this.size,
    });

    factory Tag.fromMap(Map<String, dynamic> map) {
        return Tag(
            $id: map['\$id'],
            functionId: map['functionId'],
            dateCreated: map['dateCreated'],
            command: map['command'],
            size: map['size'],
        );
    }

    Map<String, dynamic> toMap() {
        return {
            "\$id": $id,
            "functionId": functionId,
            "dateCreated": dateCreated,
            "command": command,
            "size": size,
        };
    }
}
